src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -231,21 +239,38 @@ public MedicalInventory getInventoryByReference(String reference) throws OHServi
 	/**
 	 * Verify if the object is valid for CRUD and return a list of errors, if any.
 	 *
-	 * @param medInventory
+	 * @param medicalInventory the {@link MedicalInventory} to validate
+	 * @param checkReferenceExists if {@code true}, it verifies the reference uniqueness
 	 * @throws OHDataValidationException
 	 */
-	private void validateMedicalInventory(MedicalInventory medInventory) throws OHDataValidationException {
+	private void validateMedicalInventory(MedicalInventory medicalInventory, boolean checkReferenceExists) throws OHDataValidationException {
 		List<OHExceptionMessage> errors = new ArrayList<>();
 		LocalDateTime tomorrow = LocalDateTime.now().plusDays(1);
-		String reference = medInventory.getInventoryReference();
-		if (medInventory.getInventoryDate() == null) {
+		String reference = medicalInventory.getInventoryReference();
+		if (medicalInventory.getInventoryDate() == null) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.pleaseinsertavalidinventorydate.msg")));
 		}
-		if (medInventory.getInventoryDate() != null && medInventory.getInventoryDate().isAfter(tomorrow)) {
+		if (medicalInventory.getInventoryDate() != null && medicalInventory.getInventoryDate().isAfter(tomorrow)) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.notdateinfuture.msg")));
 		}
 		if (reference == null || reference.equals("")) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.mustenterareference.msg")));
+		} else {
+			/*
+			 * Despite MINVT_REFERENCE is VARCHAR(50), this field will be copied into MMV_REFNO and concatenate to either "-charge" or "-discharge" suffixes,
+			 * hence the limit to 40. For similarity, we keep the same validation method for "ward" type inventories, despite the field would be copied into
+			 * MMVN_DESC VARCHAR(100) in those cases.
+			 */
+			if (reference.length() > 40) {
+				errors.add(new OHExceptionMessage(MessageBundle.formatMessage("angal.inventory.thereferenceistoolongmaxchars.fmt.msg", 40)));
+			}
+			if (checkReferenceExists) {
+				try {
+					checkReference(medicalInventory);
+				} catch (OHServiceException e) {
+					throw new OHDataValidationException(e.getMessages());
```

Comment:
Instead of adding the exception to `errors` which then would be caught in line 275; if the reference is too long (see line 265) the user will not see that message.   The question is how important is losing a message?

Reply:
yeah, we could use `errors.addAll(e.getMessages());` instead... or just bypass all other messages because this is more important?